docs/user-guide/v3_migration.rst

```diff
@@ -0,0 +1,204 @@
+3.0 Migration Guide
+===================
+
+Zarr-Python 3.0 introduces a number of changes to the API, including a number
+of significant breaking changes and pending deprecations.
+
+This page provides a guide highlighting the most notable changes to help you
+migrate your code from version 2 to version 3.
+
+Zarr-Python 3 represents a major refactor of the Zarr-Python codebase. Some of the
+goals motivating this refactor included:
+
+* adding support for the Zarr V3 specification (alongside the Zarr V2 specification)
+* cleaning up internal and user facing APIs
+* improving performance (particularly in high latency storage environments like
+  cloud object store)
+
+Compatibility target
+--------------------
+
+The goals described above necessitated some breaking changes to the API (hence the
+major version update), but we have attempted to maintain ~95% backwards compatibility
+in the most widely used parts of the API. This in the :class:`zarr.Array` and
+:class:`zarr.Group` classes and the "top-level API" (e.g. :func:`zarr.open_array` and
+:func:`zarr.open_group`).
+
+Getting ready for 3.0
+---------------------
+
+Ahead of the 3.0 release, we suggest projects that depend on Zarr-Python take the
```

Comment:
If the plan is to release in 6 days time, then I don't think there's much point adding a new section to the docs that says "Ahead of the 3.0 release...". I'd suggest moving the "Using zarr Python v2" section up here, and adding the pinning advice to that section. Then the flow is

1. How to continue using v2
2. How to transition to v3